### PR TITLE
feat(eks-logs): add variable to enable or disable logs flags

### PR DESCRIPTION
**Pull request by smerlos**
This pull request updates our EKS Terraform configurations to enhance flexibility for logs configuration.
**Actions taken:**
- Introduced a new variable `cluster_enabled_log_types` for dynamic log type management.
- Implemented a validation rule for the new variable to ensure only supported log types are used.
- Updated EKS example configurations to utilize the new variable, replacing hardcoded log types.
- Applied `terraform fmt --recursive` to ensure consistent code formatting.

For more details on the log types supported by Amazon EKS, please refer to the [official documentation](https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html).

**Variable Documentation:**
```hcl
variable "cluster_enabled_log_types" {
  description = "Specifies which Kubernetes control plane logs to enable. Supports 'api', 'audit', 'authenticator', 'controllerManager', and 'scheduler'. Can be left empty to disable logging."
  type        = list(string)
  default     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]

  validation {
    condition     = length(var.cluster_enabled_log_types) == 0 || alltrue([for log_type in var.cluster_enabled_log_types: contains(["api", "audit", "authenticator", "controllerManager", "scheduler"], log_type)])
    error_message = "Invalid log type. Must be one of 'api', 'audit', 'authenticator', 'controllerManager', 'scheduler', or an empty list."
  }
}
```
**QA Testing:**

The validation of the `cluster_enabled_log_types` variable was thoroughly tested to ensure its correctness. The tests were conducted using the Terraform version specified in the repository's `.tool-versions` file. It's important to note that since this update pertains to an existing configuration within the module, there was no need to recreate an entire cluster. Instead, the focus was on enhancing the way the logging flag is defined to make it more dynamic and flexible.

Specifically, the tests covered the following scenarios:
- Validation of an empty list to ensure logging is disabled when desired.
- Verification that the variable correctly accepts and processes valid log types individually and in combination.
- Confirmation that the validation rule effectively blocks invalid log type entries and provides the expected error message.

These tests ensured that the validation of the `cluster_enabled_log_types` variable functions as intended, guaranteeing accurate and reliable log type configuration.

### CODE REVIEW

File: modules/eks/variables.tf [hunk 1]
```diff
@@ -205,3 +205,14 @@ variable "ssh_to_nodes_allowed_cidr_blocks" {
...
   default     = null
 }
+
+variable "cluster_enabled_log_types" {
+  description = "List of log types that will be enabled for the EKS cluster. Can be a subset of ['api', 'audit', 'authenticator', 'controllerManager', 'scheduler'] or an empty list."
+  type        = list(string)
+  default     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+
+  validation {
+    condition     = length(var.cluster_enabled_log_types) == 0 || alltrue([for val in var.cluster_enabled_log_types : contains(["api", "audit", "authenticator", "controllerManager", "scheduler"], val)])
```
Comment (g-iannelli) [inline modules/eks/variables.tf:215]:
```suggestion
        condition     = length(coalesce(var.cluster_enabled_log_types, [])) == 0 || alltrue([for val in coalesce(var.cluster_enabled_log_types, []) : contains(["api", "audit", "authenticator", "controllerManager", "scheduler"], val)])
```

Comment (g-iannelli) [reply modules/eks/variables.tf:215]: To support `null` as value

**Review by g-iannelli — CHANGES_REQUESTED** [review]

File: modules/eks/variables.tf (hunk 1 above)
Comment (smerlos) [reply modules/eks/variables.tf:215]: According to the documentation, the default value for the array ([]) is an empty one. So, I am unsure whether we should also accept null values. As per the validation rules, we are only allowed to accept an empty array (as described in the providers' default) and the items that are allowed as per the documentation. @g-iannelli

**Review by g-iannelli — APPROVED** [review]
LGTM

File: modules/eks/variables.tf [hunk 2]
```diff
@@ -205,3 +205,13 @@ variable "ssh_to_nodes_allowed_cidr_blocks" {
...
+variable "cluster_enabled_log_types" {
+  description = "List of log types that will be enabled for the EKS cluster. Can be a subset of ['api', 'audit', 'authenticator', 'controllerManager', 'scheduler'] or an empty list."
+  type        = list(string)
+  default     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+
+  validation {
+    condition     = length(coalesce(var.cluster_enabled_log_types, [])) == 0 || alltrue([for val in coalesce(var.cluster_enabled_log_types, []) : contains(["api", "audit", "authenticator", "controllerManager", "scheduler"], val)])
+    error_message = "The log type must be one of the following: api, audit, authenticator, controllerManager, scheduler, or the list must be empty."
+  }
+}
```
Comment (omissis) [inline modules/eks/variables.tf:217]: nitpick: missing NL at the EOF

```suggestion
}

```

**Review by omissis — DISMISSED** [review]
LGTM, this will likely need two follow up PRs on fury-distribution and furyctl

**Review by g-iannelli — CHANGES_REQUESTED** [review]
I found a better solution whereby in case of `null` value, the default one is used.

File: modules/eks/variables.tf [hunk 3]
```diff
@@ -205,3 +205,13 @@ variable "ssh_to_nodes_allowed_cidr_blocks" {
...
   default     = null
 }
+variable "cluster_enabled_log_types" {
+  description = "List of log types that will be enabled for the EKS cluster. Can be a subset of ['api', 'audit', 'authenticator', 'controllerManager', 'scheduler'] or an empty list."
+  type        = list(string)
+  default     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+
+  validation {
+    condition     = length(coalesce(var.cluster_enabled_log_types, ["api", "audit", "authenticator", "controllerManager", "scheduler"])) == 0 || alltrue([for val in coalesce(var.cluster_enabled_log_types, ["api", "audit", "authenticator", "controllerManager", "scheduler"]) : contains(["api", "audit", "authenticator", "controllerManager", "scheduler"], val)])
+    error_message = "The log type must be one of the following: api, audit, authenticator, controllerManager, scheduler, or the list must be empty."
```
[the remark this answers is not in the document]
Comment (smerlos) [reply modules/eks/variables.tf:215]: Thanks, the nullable was implemented.

**Review by g-iannelli — APPROVED** [review]
LGTM

**Pull request merged**